Product/ProductPriceManager.php

```diff
@@ -79,6 +79,28 @@ public function getBasePriceForCurrency(ProductInterface $product, $currency = n
     }
 
     /**
+     * Returns the special price for the product by a given currency
+     *
+     * @param ProductInterface $product
+     * @param null|string $currency
+     *
+     * @return null|\Sulu\Bundle\ProductBundle\Entity\ProductPrice
+     */
+    public function getSpecialPriceForCurrency(ProductInterface $product, $currency = null)
+    {
+        $currency = $currency ?: $this->defaultCurrency;
+        if ($prices = $product->getSpecialPrices()) {
+            foreach ($prices as $price) {
+                if ($price->getCurrency()->getCode() == $currency) {
+                    return $price;
```

Comment:
are you checking somewhere if special-price is still valid (start and end-date) ?